### ISSUE TITLE
Installation failing on remote server

### ISSUE THREAD

**Issue by elfico** [nopSolutions/nopCommerce#2970]
nopCommerce version: 4.0 (latest build)

Steps to reproduce the problem: 
Server: Windows Server 2016 on AWS EC2 Elastic Beanstalk.
Sql Server Express on AWS RDS

1. Created an environment on AWS EC2 Elastic Beanstalk
2. Uploaded NopCommerce 4.0 and deployed.
3. Filled all forms on the installation page and Clicked install
4. Installation starts and after a few minutes, stops with error "**Setup failed: Value cannot be null. Parameter name: hashAlgorithm**"

After the failed installation, I noticed that the database was created. Subsequent attempt to install failed.

AppData folder contains a dataSettings.json  whose content is
```
{
  "DataProvider": null,
  "DataConnectionString": null,
  "RawDataSettings": {}
}
```
Showing that the dataSettings.json was not properly created during installation. I modified the dataSettings.json as follows :
```
{
  "DataProvider": "sqlserver",
  "DataConnectionString": "Data Source=servername;Initial Catalog=dbname;Integrated Security=False;Persist Security Info=False;User ID=userid;Password=password",
  "RawDataSettings": {}
}
```
And reloaded the page and some parts opened with no css, js, and very unstructured html

Installation on local machine works fine without any issues.

I tried to connect to install on local machine connecting to remote db but results were same. I have searched over and there seemed to be no fix or workaround for this.

**Comment by AndreiMaz** [comment]
the issue is that it timed out

**Comment by VR-Architect** [comment]
I had this error too:
https://github.com/nopSolutions/nopCommerce/issues/3162

**Comment by pantonis** [comment]
@AndreiMaz Timed out? can you tell more details? Thanks

**Comment by sharpapp-mareklanduch** [comment]
There is a bug in sample data, I installed nopCommerce without sample data and everything is working now.

**Comment by mkgn** [comment]
Timeout may not be the only reason. Here are my findings.

1) I installed nopCommerce using the Azure Marketplace package. I didn't setup test data. Installation went fine and site works.

2) I got the latest code from GitHub (v 4.1), compiled and pushed to Azure. When the installation page appear, I used the same option (setup without sample data) and it consistently give this error "Setup failed: Value cannot be null. Parameter name: hashAlgorithm". As mentioned by @elfico database has tables (not sure whether its partial db), dataSettings.json has empty data. As of now, I can keep on reproducing this ALL the time.

So I guess something else is going wrong. If it's timeout issue then it should be the case when installing using Azure NopCommerce marketplace package as well.
